sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3137,24 +3137,73 @@ SIREPO.app.directive('sbatchLoginModal', function() {
 
 });
 
-SIREPO.app.directive('sbatchCoresAndHours', function(appState) {
+SIREPO.app.directive('sbatchOptions', function(appState) {
     return {
         restrict: 'A',
         scope: {
-            simState: '=sbatchCoresAndHours',
+            simState: '=sbatchOptions',
         },
         template: [
             '<div class="clearfix"></div>',
             '<div style="margin-top: 10px" data-ng-show="showCoresAndHours()">',
                 '<div data-model-field="\'sbatchHours\'" data-model-name="simState.model" data-label-size="3" data-field-size="3"></div>',
                 '<div data-model-field="\'sbatchCores\'" data-model-name="simState.model" data-label-size="3" data-field-size="3"></div>',
+                '<div data-model-field="\'sbatchQueue\'" data-model-name="simState.model" data-label-size="3" data-field-size="3"  data-ng-click="sbatchQueueFieldIsDirty = true"></div>',
             '</div>',
         ].join(''),
-        controller: function($scope) {
+        controller: function($scope, $element) {
+            $scope.sbatchQueueFieldIsDirty = false;
+            var model = appState.models[$scope.simState.model];
+            var coresPerNode = 32;
+            var max = {
+                Hours: {
+                    debug: 0.4,
+                    regular: 47
+                },
+                Cores: {
+                    debug: 62 * coresPerNode,
+                    regular: 1932 * coresPerNode
+                }
+            };
+
+            function setHoursAndCores() {
+                if (! (model.sbatchQueue in max.Hours)) {
+                    return;
+                }
+                ['Hours', 'Cores'].forEach(function(e) {
+                    model['sbatch' + e] = Math.min(
+                        model['sbatch' + e],
+                        max[e][model.sbatchQueue]
+                    );
+                });
+            }
+
+            function setQueue() {
+                if (model.sbatchHours <= max.Hours.debug && model.sbatchCores <= max.Cores.debug) {
+                    model.sbatchQueue = 'debug';
+                    return;
+                }
+                if (model.sbatchHours <= max.Hours.regular && model.sbatchCores <=  max.Cores.regular) {
+                    model.sbatchQueue = 'regular';
+                    return;
+                }
+                model.sbatchQueue = 'realtime';
```

Comment:
This is not necessarily true. We should not "trim" to realtime, because it is a special queue. If the user requests too many cores or hours, we should simply limit them to the max.

Reply:
https://github.com/radiasoft/sirepo/pull/2511/commits/fb55c2011646a1e6d9593bfb0c069e4c4223b959